### PR TITLE
Add integration tests for the containers.

### PR DESCRIPTION
**Pull request by nadiaya**
Add integration tests for the containers.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Test args for pytest
+[tool:pytest]
+addopts =
+    -vv
+    -s
```
Comment (laurenyu) [inline setup.cfg]: add a newline to the end

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (laurenyu) [inline setup.py]: should this just be 2018?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        "Development Status :: 5 - Production/Stable",
+        "Intended Audience :: Developers",
+        "Natural Language :: English",
+        "License :: OSI Approved :: Apache Software License",
+        "Programming Language :: Python",
+        'Programming Language :: Python :: 3.6',
+    ],
+
+    install_requires=['numpy', 'tox', 'flake8', 'pytest', 'pytest-cov', 'pytest-xdist', 'mock',
+                      'sagemaker', 'docker-compose'],
```
Comment (laurenyu) [inline setup.py]: should these be extra requires for test only?

File: test/conftest.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+logger = logging.getLogger(__name__)
+logging.getLogger('boto').setLevel(logging.INFO)
+logging.getLogger('botocore').setLevel(logging.INFO)
+logging.getLogger('factory.py').setLevel(logging.INFO)
+logging.getLogger('auth.py').setLevel(logging.INFO)
+logging.getLogger('connectionpool.py').setLevel(logging.INFO)
+
+SCRIPT_PATH = os.path.dirname(os.path.realpath(__file__))
+LATEST_COACH_VERSION = '0.11.0'
+LATEST_RAY_VERSION = '0.5.3'
```
Comment (laurenyu) [inline test/conftest.py]: can these be read from somewhere instead?

File: test/conftest.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+
+def pytest_addoption(parser):
+    parser.addoption('--base-name', default='sagemaker-rl')
```
Comment (laurenyu) [inline test/conftest.py]: should we make this consistent with other repos, i.e. ``--docker-base-name``?

File: test/conftest.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    parser.addoption('--framework', default='tensorflow')
+    parser.addoption('--region', default='us-west-2')
+    parser.addoption('--toolkit', default='coach')
```
Comment (laurenyu) [inline test/conftest.py]: seems a little weird to me to set defaults for `framework` and `toolkit` - why did you choose the ones you did?

File: test/integration/__init__.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+RESOURCE_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'resources'))
+
+MODEL_SUCCESS_FILES = {
+    'output': ['success'],
+    'model': ['model-symbol.json', 'model-shapes.json', 'model-0000.params'],
+}
+
+# Workaround for the intermittent worker timeout errors
+# TODO: find and solve the root cause of this issue
+NUM_MODEL_SERVER_WORKERS = 2
```
Comment (laurenyu) [inline test/integration/__init__.py]: 😭

File: test/integration/local/test_ray.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    estimator = RLEstimator(entry_point=cartpole,
+                            source_dir=source_dir,
+                            role='SageMakerRole',
+                            train_instance_count=1,
+                            train_instance_type=local_instance_type,
+                            sagemaker_session=sagemaker_local_session,
+                            output_path='file://{}'.format(tmpdir),
+                            image_name=docker_image)
+
+    estimator.fit()
```
Comment (laurenyu) [inline test/integration/local/test_ray.py]: why do only some of these fit calls have the timeout?

File: test/resources/coach_cartpole/mxnet_deploy.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
+import os
+import json
+import mxnet as mx
```
Comment (laurenyu) [inline test/resources/coach_cartpole/mxnet_deploy.py]: newline between ll. 2 and 3. also the mxnet imports should be alphabetized

File: setup.py (hunk 3 above)
Comment (nadiaya) [reply setup.py]: there is nothing except tests right now... so maybe if we add some code then?

File: test/conftest.py (hunk 6 above)
Comment (nadiaya) [reply test/conftest.py]: because 6 out of 8 containers are coach and tensorflow :)

File: test/integration/__init__.py (hunk 7 above)
Comment (nadiaya) [reply test/integration/__init__.py]: oops

File: test/integration/local/test_ray.py (hunk 8 above)
Comment (nadiaya) [reply test/integration/local/test_ray.py]: local mode don't need them, right?

File: test/conftest.py (hunk 4 above)
Comment (nadiaya) [reply test/conftest.py]: not really

File: test/conftest.py [hunk 10]
```diff
@@ -35,7 +35,7 @@
 
 
 def pytest_addoption(parser):
-    parser.addoption('--base-name', default='sagemaker-rl')
+    parser.addoption('--docker-base-name', default=None)
```
Comment (laurenyu) [inline test/conftest.py]: is "default=None" necessary?

File: test/conftest.py (hunk 6 above)
Comment (laurenyu) [reply test/conftest.py]: but not in that combination :P

Comment (nadiaya) [reply test/conftest.py]: fine ,you convinced me :)

**Pull request merged**